packages/firestore/src/core/sync_engine.ts

```diff
@@ -958,9 +964,42 @@ export class SyncEngine implements RemoteSyncer, SharedClientStateSyncer {
           targetId,
           /*current=*/ false
         );
+      } else {
+        // For queries that have a local View, we need to update their state
+        // in LocalStore (as the resume token and the snapshot version
+        // might have changed) and reconcile their views with the persisted
+        // state (the list of syncedDocuments may have gotten out of sync).
+        await this.localStore.releaseTarget(
+          targetId,
+          /*keepPersistedQueryData=*/ true
+        );
+
+        assert(
```

Comment:
If I am not mistaken, since assert should never trigger since it matches the if-check in line 952? Is it meant to guard against the `releaseQuery` call above?